llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -767,6 +767,142 @@ Operation *FftOp::getTiledImplementation(OpBuilder &builder, ValueRange outputs,
   return tiledFftOp;
 }
 
+//===----------------------------------------------------------------------===//
+// ScanOp
+//===----------------------------------------------------------------------===//
+
+static LogicalResult verifyScanOp(ScanOp op) {
+  if (op.inputs().size() != 1) {
+    return op.emitOpError("expected one input operand");
+  }
+  if (op.outputs().size() != 1) {
+    return op.emitOpError("expected one output operand");
+  }
+  return success();
+}
+
+SmallVector<Range> ScanOp::getIterationDomain(OpBuilder &builder) {
+  int64_t operandRank = getOperandRank();
+  SmallVector<Range> loopBounds(operandRank);
+  Location loc = getLoc();
+  Value zero = builder.create<arith::ConstantIndexOp>(loc, 0);
+  Value one = builder.create<arith::ConstantIndexOp>(loc, 1);
+  Value source = input();
+  for (auto dim : llvm::seq<int64_t>(0, operandRank)) {
```

Comment:
Wont the number of iterations for the scan dimension change if this an inclusive or exclusive scan? Depends on how you would design this op and how it lowers to loops.

Reply:
You are right. My bad. I was Thinking that for the exclusive scan the size of the result is 1 greater than the size of the input to hold the value of the last element. Seems like thats not the case. So this is good.